db/openshift.deploy.yml

```diff
@@ -326,3 +329,99 @@ objects:
               dnsPolicy: "ClusterFirst"
               serviceAccountName: "${JOB_SERVICE_ACCOUNT}"
               serviceAccount: "${JOB_SERVICE_ACCOUNT}"
+  - kind: CronJob
+    apiVersion: "batch/v1"
+    metadata:
+      name: ${NAME}-${ZONE}-${COMPONENT}-restore
+      labels:
+        app: ${NAME}-${ZONE}
+        cronjob: ${NAME}-${ZONE}
+    spec:
+      schedule: "0 0 31 2 *" # At 00:00 on day-of-month 31 in February, so it won't run, we'll trigger it manually when need
+      concurrencyPolicy: "Replace"
+      successfulJobsHistoryLimit: ${{SUCCESS_JOBS_HISTORY_LIMIT}}
+      failedJobsHistoryLimit: ${{FAILED_JOBS_HISTORY_LIMIT}}
+      jobTemplate:
+        metadata:
+          labels:
+            app: ${NAME}-${ZONE}
+            cronjob: ${NAME}-${ZONE}
+        spec:
+          backoffLimit: 0
+          template:
+            metadata:
+              labels:
+                app: ${NAME}-${ZONE}
+                cronjob: ${NAME}-${ZONE}
+            spec:
+              containers:
+                - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                  # use the same image as our database, so we can run the psql command
+                  image: image-registry.apps.silver.devops.gov.bc.ca/${OC_NAMESPACE}/${NAME}-${ZONE}-${COMPONENT}:${ZONE}-db
+                  command: ["/bin/sh", "-c"]
+                  args:
+                  - |
+                    sql_file=$(find backups/${BACKUP_FILE_DIR} -type f -name "*.sql" -print0)
+                    if [  -z $sql_file ]; then
+                      sql_zip_file=$(find backups/${BACKUP_FILE_DIR} -type f -name "*.sql.gz" -print0)
+                      if [ -z $sql_zip_file ]; then
+                        echo "No zipped backup file found"
+                      else
+                        echo "Found zipped backup file: $sql_zip_file"
+                        gzip -dk $sql_zip_file
+                      fi
+                    fi
+                    sql_file=$(find backups/${BACKUP_FILE_DIR} -type f -name "*.sql" -print0)
+                    if [  -z $sql_file ]; then
+                      echo "Error: No backup SQL file is found"
+                      exit 1
+                    else
+                      echo "Found SQL file, rename existing database, and create a new empty database"
+                      psql -h ${NAME}-${ZONE}-${COMPONENT} -U ${POSTGRES_USER} -c "DROP DATABASE IF EXISTS ${OLD_FOM_DATABASE_NAME};" -c "ALTER DATABASE fom RENAME TO ${OLD_FOM_DATABASE_NAME};" -c "CREATE DATABASE fom;"
```

Comment:
The process documented in the wiki could talk more about what to do if the restore fails (or wasn't the correct restore point and needs to be redone), which could mean picking a different value for old_fom_database_name, and/or one option for reverting the restore would be to drop the fom database and rename old_fom to fom...

Reply:
Basic reverting restore failure is included in adjusted script.